src/pages/home/report/ParticipantLocalTime.js

```diff
@@ -16,60 +16,52 @@ const propTypes = {
     ...withLocalizePropTypes,
 };
 
-class ParticipantLocalTime extends PureComponent {
-    constructor(props) {
-        super(props);
-        this.getParticipantLocalTime = this.getParticipantLocalTime.bind(this);
-        this.state = {
-            localTime: this.getParticipantLocalTime(),
-        };
+function getParticipantLocalTime(participant, preferredLocale) {
+    const reportRecipientTimezone = lodashGet(participant, 'timezone', CONST.DEFAULT_TIME_ZONE);
+    const reportTimezone = DateUtils.getLocalMomentFromDatetime(preferredLocale, null, reportRecipientTimezone.selected);
+    const currentTimezone = DateUtils.getLocalMomentFromDatetime(preferredLocale);
+    const reportRecipientDay = reportTimezone.format('dddd');
+    const currentUserDay = currentTimezone.format('dddd');
+
+    if (reportRecipientDay !== currentUserDay) {
+        return `${reportTimezone.format('LT')} ${reportRecipientDay}`;
     }
+    return `${reportTimezone.format('LT')}`;
+}
+
+function ParticipantLocalTime(props) {
+    const {participant, preferredLocale, translate} = props;
```

Comment:
Are you suggesting I shouldn't destructure `translate` because that is not watched and we can keep accessing it as `props.translate`?

If that is the case, I don't have strong feeling toward that, but I think the best would be to have access to the `translate` function through a hook like it was suggested here: https://expensify.slack.com/archives/C01GTK53T8Q/p1685068222924839?thread_ts=1684969645.381109&cid=C01GTK53T8Q